scripts/cli_options.py

```diff
@@ -53,21 +63,144 @@ def add_common_options(parser):
              "Available options: {}"
-             .format(order_choose_algorithms[0], order_choose_algorithms),
+             .format('random_under_max_order_choose',
+                     ', '.join(order_choose_algorithms.keys())),
         dest='order_choose_fn')
+    add_order_choose_short_options(parser)
+
+
+def add_order_choose_short_options(parser):
+    for name in sorted(order_choose_algorithms.keys()):
+        option = order_choose_algorithms[name]
+        parser.add_option(
+            option,
+            help="alias for --order-choose-algorithm={}".format(name),
+            nargs=0,
+            action='callback',
+            callback=get_order_choose_algorithm,
+            callback_kwargs={'value_kw': name},
+            dest='order_choose_fn')
 
 
-def get_order_choose_algorithm(option, opt_str, value, parser):
+def get_order_choose_algorithm(option, opt_str, value, parser, value_kw=None):
+    value = value_kw or value
     if value not in order_choose_algorithms:
-        raise OptionValueError("{} must be one of {}"
-                               .format(opt_str, order_choose_algorithms))
+        raise OptionValueError("{} must be one of {}".format(
+                opt_str, list(order_choose_algorithms.keys())))
     fn = getattr(jmclient.support, value, None)
     if not fn:
         raise OptionValueError("internal error: '{}' order choose algorithm not"
                                " found".format(value))
     setattr(parser.values, option.dest, fn)
 
 
+def get_max_cj_fee_values(config, parser_options):
+    CONFIG_SECTION = 'POLICY'
+    CONFIG_OPTION = 'max_cj_fee_'
+    # rel, abs
+    fee_values = [None, None]
+    fee_types = [float, int]
+
+    for i, option in enumerate(('rel', 'abs')):
+        value = getattr(parser_options, CONFIG_OPTION + option, None)
+        if value is not None:
+            fee_values[i] = fee_types[i](value)
+            continue
+
+        try:
+            value = config.get(CONFIG_SECTION, CONFIG_OPTION + option)
+        except NoOptionError:
+            continue
+
+        if value is not None:
+            fee_values[i] = value
+
+    if len(filter(lambda x: x is None, fee_values)):
+        fee_values = prompt_user_for_cj_fee(*fee_values)
+
+    return tuple(map(lambda j: fee_types[j](fee_values[j]),
+                     range(len(fee_values))))
+
+
+def prompt_user_for_cj_fee(rel_val, abs_val):
+    msg = """Joinmarket will choose market makers randomly as long as their
+fees are below a certain maximum value, or percentage. The suggested maximums are:
+
+X = {rel_val:%}
+Y = {abs_val} satoshis
+
+Those values were chosen randomly for you (unless you already set one of them
+in your joinmarket.cfg or via a CLI option).
+
+Since you are using N counterparties, if you agree to these values, your
+**maximum** coinjoin fee will be either N*Y satoshis or N*X percent of your
+coinjoin amount C, which is Z satoshis, depending on which is larger. The
```

Comment:
I actually intended these values N*X , N*Y, C and Z to be filled in for the user here.

Reply:
as discussed on IRC, this could only possibly be done for sendpayment, not tumbler